sequencer/sequencer.go

```diff
@@ -175,6 +183,166 @@ func (s *Sequencer) Start(ctx context.Context) {
 	<-ctx.Done()
 }
 
+func (s *Sequencer) updateDataStreamerFile(ctx context.Context, streamServer *datastreamer.StreamServer) {
+	var currentL2Block uint64
+	var currentTxIndex uint64
+	var err error
+
+	header := streamServer.GetHeader()
+
+	if header.TotalEntries == 0 {
+		// Get Genesis block
+		genesisL2Block, err := s.state.GetDSGenesisBlock(ctx, nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = streamServer.StartAtomicOp()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		genesisBlock := state.DSL2BlockStart{
+			BatchNumber:    genesisL2Block.BatchNumber,
+			L2BlockNumber:  genesisL2Block.L2BlockNumber,
+			Timestamp:      genesisL2Block.Timestamp,
+			GlobalExitRoot: genesisL2Block.GlobalExitRoot,
+			Coinbase:       genesisL2Block.Coinbase,
+			ForkID:         genesisL2Block.ForkID,
+		}
+
+		log.Infof("Genesis block: %+v", genesisBlock)
+
+		_, err = streamServer.AddStreamEntry(1, genesisBlock.Encode())
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		genesisBlockEnd := state.DSL2BlockEnd{
+			L2BlockNumber: genesisL2Block.L2BlockNumber,
+			BlockHash:     genesisL2Block.BlockHash,
+			StateRoot:     genesisL2Block.StateRoot,
+		}
+
+		_, err = streamServer.AddStreamEntry(state.EntryTypeL2BlockEnd, genesisBlockEnd.Encode())
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = streamServer.CommitAtomicOp()
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		latestEntry, err := streamServer.GetEntry(header.TotalEntries - 1)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		log.Infof("Latest entry: %+v", latestEntry)
+
+		switch latestEntry.EntryType {
+		case state.EntryTypeL2BlockStart:
+			log.Info("Latest entry type is L2BlockStart")
+			currentL2Block = binary.LittleEndian.Uint64(latestEntry.Data[8:16])
+		case state.EntryTypeL2Tx:
+			log.Info("Latest entry type is L2Tx")
+			for latestEntry.EntryType == state.EntryTypeL2Tx {
+				currentTxIndex++
+				latestEntry, err = streamServer.GetEntry(header.TotalEntries - currentTxIndex)
+				if err != nil {
+					log.Fatal(err)
+				}
+			}
+			if latestEntry.EntryType != state.EntryTypeL2BlockStart {
+				log.Fatal("Latest entry is not a L2BlockStart")
+			}
+			currentL2Block = binary.LittleEndian.Uint64(latestEntry.Data[8:16])
+		case state.EntryTypeL2BlockEnd:
+			log.Info("Latest entry type is L2BlockEnd")
+			currentL2Block = binary.LittleEndian.Uint64(latestEntry.Data[0:8])
+		}
+	}
+
+	log.Infof("Current transaction index: %d", currentTxIndex)
+	if currentTxIndex == 0 {
+		currentL2Block++
+	}
+	log.Infof("Current L2 block number: %d", currentL2Block)
+
+	var limit uint64 = 1000
```

Comment:
I don't know if makes sense add a config parameter to specify this limit (chunk)